src/semidiscretization/semidiscretization.jl

```diff
@@ -180,6 +180,11 @@ function compute_coefficients!(u_ode, func, t, semi::AbstractSemidiscretization)
     compute_coefficients!(backend, u, func, t, mesh_equations_solver_cache(semi)...)
 end
 
+# Required for `VolumeIntegralSubcellLimiting` volume integral
+function Base.resize!(semi::AbstractSemidiscretization, new_size)
+    resize!(semi, semi.solver.volume_integral, new_size)
+end
+
```

Comment:
We could also remove this at all and directly call `resize!` with `(semi, volume_integral, new_size)`. Maybe that's also clearer since you see that it depends on the volume integral. What do you think? Should I prepare the necessary changes?
```suggestion
```